src/compiler/checker.rs

```diff
@@ -634,7 +646,11 @@ impl TypeChecker {
         result
     }
 
-    fn get_type_names_for_error_display(&self, left: &Type, right: &Type) -> (String, String) {
+    fn get_type_names_for_error_display(
+        &self,
+        left: Rc<Type>,
```

Comment:
Some method called from here needed to return an `Rc<Type>` so until I implement the same pattern (presumably with the help of a macro) for `Type`'s as I did for `Node`'s (so that I can eg call `.type_wrapper()` to get an `Rc<Type>` from a `&TypeInterface`) the `Rc`-ness needs to recursively "infect" the callers